replica/database.hh

```diff
@@ -1001,8 +1001,8 @@ public:
         return _pending_flushes_phaser.advance_and_await();
     }
 
-    future<> await_pending_ops() noexcept {
-        return when_all(await_pending_reads(), await_pending_writes(), await_pending_streams(), await_pending_flushes()).discard_result();
+    future<> close_for_read_write_ops() noexcept {
+        return when_all(_pending_reads_phaser.close(), _pending_writes_phaser.close(), _pending_streams_phaser.close()).discard_result();
```

Comment:
I wonder if `table::stop()` should stop the table views first.
The view builder has a bacground loop that calls `advance_and_await` on the write and streams phasers.
This tries to close the phaser's current gate and will assert if it's already closed.

Regardless, `advance_and_await` should fail more gracefully now that we support phased_barrier::close(), so if it's already closed it should throw a gate_closed exception rather than fail the assert,

Reply:
I don't want to reshuffle the order of stops in this series. Will make the change you requested to `advance_and_await()` instead.